taxonomy/tests/test_api.py

```diff
@@ -0,0 +1,346 @@
+                              order_by('name')))
+
+    def test_create_term(self):
+        """
+        Test create_term
+        """
+        self.assertEquals([self.term],
+                          list(Term.objects.all()))
+
+        term = create_term(self.vocabulary.id, "label", 9)
+        self.assertEquals([self.term, term],
+                          list(Term.objects.order_by('id')))
+
+        self.assertRaises(Vocabulary.DoesNotExist,
+                          lambda: create_term(456, "label", 9))
+        self.assertRaises(IntegrityError,
+                          lambda: create_term(self.vocabulary.id, None, 9))
+        create_term(self.vocabulary.id, "label2", 99.7)
+        # term weight is truncated after being stored in database
+        self.assertEquals(1, Term.objects.filter(weight=99).count())
+
+    def test_create_vocabulary(self):
+        """
+        Test create_vocabulary
+        """
+        self.assertEquals([self.vocabulary],
+                          list(Vocabulary.objects.all()))
+
+        v = create_vocabulary(self.repository.id,
+                              "vocabulary",
+                              "description",
+                              False,
+                              "f",
+                              9)
+        self.assertEquals([self.vocabulary, v],
+                          list(Vocabulary.objects.order_by('id')))
+        self.assertRaises(Repository.DoesNotExist,
+                          lambda: create_vocabulary(9,
+                                                    "vocabulary",
+                                                    "description",
+                                                    False,
+                                                    "m",
+                                                    9))
+
+        # int is cast to string
+        create_vocabulary(self.repository.id,
+                          9,
+                          "description",
+                          False,
+                          "f",
+                          9)
+        self.assertEquals("9", Vocabulary.objects.order_by('id').last().name)
+
+        create_vocabulary(self.repository.id,
+                          "name",
+                          9,
+                          False,
+                          "m",
+                          9)
+        self.assertEquals("9", Vocabulary.objects.order_by('id').
+                          last().description)
+
+        # "abc" is cast to boolean True
+        create_vocabulary(self.repository.id,
+                          "vocabulary",
+                          "description",
+                          "abc",
+                          "f",
+                          9)
+        self.assertEquals(True, Vocabulary.objects.order_by('id').last().
+                          required)
+
+        # pass in None, violates constraints
+        self.assertRaises(IntegrityError,
```

Comment:
If `create_vocabulary` were using `full_clean` it would be raising a `ValidationError` instead. Wouldn't it still be worth asserting how it handles `None`?